packages/tooling/fast-tooling/src/data-utilities/web-component.ts

```diff
@@ -0,0 +1,80 @@
+import { DataType } from "./types";
+
+export interface WebComponentAttribute {
+    /**
+     * The name of the attribute
+     */
+    name: string;
+
+    /**
+     * The type of value for this attribute
+     */
+    type: DataType;
+
+    /**
+     * The description of this attribute
+     */
+    description: string;
+
+    /**
+     * The default value of this attribute
+     */
+    default: string;
+
+    /**
+     * Whether this was attribute is required
+     */
+    required: boolean;
+}
+
+export interface WebComponentSlot {
+    /**
+     * The name of the slot, if this is the default slot
+     * this should be an emptry string
+     */
+    name: string;
+
+    /**
+     * The description of use of this slot
+     */
+    description: string;
+}
+
+export interface WebComponentDefinitionTag {
+    /**
+     * The name of the tag
+     */
+    name: string;
+
+    /**
+     * The description of this web component tag
+     */
+    description: string;
+
+    /**
```

Comment:
Does this definition purposefully omit a "property" vs "attribute" distinction?

Reply:
For now properties are not defined, there is intended to be a distinction between "property" and "attribute". This definition was taken from an ongoing discussion into what this definition should look like.

See: https://github.com/webcomponents/custom-elements-json/issues/3#issuecomment-546750767

Reply:
@janechu Does this light up VS intellisense automatically if we put the generated json file into our package or do we need to do some other work for that?

Reply:
There's some work involved, I haven't investigated fully but this is very close to the definition needed for VS Code web components plugin referenced in those issues. It is incomplete, I only included what the tooling needs for now with the intent to expand it later.

Reply:
@janechu What do you feel our chances are of getting that generated in time for release? I don't know how your priority list looks so wasn't sure if it would fit in. A friend of mine tried out the components yesterday and noted how much improved he felt the dev experience would be if we had that on launch. He had used some other libraries that had that, so he had a point of reference.

Reply:
Does this need to be generated? Could we write them manually for release?

Also - could you add the above link as a reference the the code-file? I think that is important context for these interfaces

Reply:
Definitely *could* be hand-written if needed and I'd be willing to do that if needed for sure.

Reply:
I think it would be faster to write them, it shouldn't be that and I've got a pretty hefty backlog to get through for our release date.